### PR TITLE
add support for indexing SAR data into elasticsearch re-work

### PR DESCRIPTION
**Pull request by arcolife**
refs: 
- issue details - https://github.com/distributed-system-analysis/pbench/pull/176#issuecomment-287229824
- reason for closing #176 in favour of this one - https://github.com/distributed-system-analysis/pbench/pull/176#issue-137501232

### CODE REVIEW

File: server/pbench/bin/index-pbench [hunk 1]
```diff
@@ -1160,7 +1160,7 @@ def mk_user_specified_metadata(options):
 
 def mk_metadata(fname, tb, mdconf, md5sum):
     """Return a dict with metadata about a tarball"""
-    mtime = datetime.fromtimestamp(os.stat(fname)[stat.ST_MTIME])
```
Comment (portante) [inline server/pbench/bin/index-pbench]: Can you submit just this change as a separate PR before the other change?

File: server/pbench/bin/gold/test-7.8.txt [hunk 2]
```diff
@@ -1039,7 +1039,19 @@ len(actions) = 19
         },
         "_type": "pbench-tool-data-iostat"
     }
-]
+]could not convert binary: /var/tmp/pbench-test-server/tmp/uperf__2016-10-06_16:34:03/1-tcp_stream-16384B-32i/sample1/tools-default/flat8_40gb/sar/sar.data
```
Comment (portante) [inline server/pbench/bin/gold/test-7.8.txt]: We need to resolve this issues.  Are you expecting to have test sar data made available in the unit test for this one?

**Comment by portante** [comment]
And then close this PR, actually and use the other one since we ended up merging the wip-indexing branch to master?

File: server/pbench/bin/index-pbench (hunk 1 above)
Comment (arcolife) [inline server/pbench/bin/index-pbench]: https://github.com/distributed-system-analysis/pbench/pull/528

File: server/pbench/bin/gold/test-7.8.txt (hunk 2 above)
Comment (arcolife) [inline server/pbench/bin/gold/test-7.8.txt]: so there's a separate test-11 that is evaluated to check sar-index work. uperf test contains sar data and that is being detected as a part of the new index-pbench code that was pushed in this PR. Either we should specify a flag (--disable-sar ?) to index pbench or we should remove this data in uperf. Removing it implies sar is not really a part of uperf results (but which it is!). But for unit tests, it sounds reasonable and IIUC that's what you meant here?

**Comment by arcolife** [comment]
So this PR is dependent on presence of https://github.com/distributed-system-analysis/pbench/pull/528 in master for its tests to pass locally. Shall we merge that first so then I could rebase this work again (per se, in #176)?

**Comment by arcolife** [comment]
anyway, I've rebased against new master, also against utc-fix (#528) so we're good.
